tap_wikipedia/models/config.py

```diff
@@ -0,0 +1,38 @@
+from pathlib import Path
+from typing import Annotated
+
+from appdirs import user_cache_dir
+from pydantic import BaseModel, Field, field_validator
+
+from tap_wikipedia.models.types import EnrichmentType, SubsetSpecification
+
+
+class Config(BaseModel):
+    """A Pydantic Model to hold configuration values of tap-wikipedia."""
+
+    abstracts_dump_url: Annotated[
+        str,
+        Field(
+            min_length=1,
+            default="https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-abstract.xml.gz",
+            validation_alias="abstracts-dump-url",
+        ),
+    ]
+    cache_directory_path: Annotated[
+        Path,
+        Field(
+            default=Path(user_cache_dir("abstracts", "tap-wikipedia")),
+            validation_alias="cache_directory_path",
+        ),
+    ]
+    enrichments: tuple[EnrichmentType, ...] | None = None
+    subset_specifications: Annotated[
+        tuple[SubsetSpecification, ...] | None,
+        Field(validation_alias="subset-specifications"),
```

Comment:
comment: I'm not familiar with this syntax. I like it better than what I usually do, 

```
subset_specifications: tuple[SubsetSpecification, ...] | None = Field(None, xxx)
```